### DISCUSSION TITLE
How to return response from Elsa workflow before completion?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#1666] [Q&A]
Hi @sfmskywalker 
Here is my use case that I want to implement:

- I have a parent workflow that calls the child workflow. The parent workflow's **RunWorkflow** activity mode is **blocking**. I need to associate the current parent workflow instance with my own object Id to create an association to view the status against that associated field. Is there any mechanism in Elsa that I may return the workflow instance  Id back and the remaining workflow will complete its execution?
**Note:** I don't have any blocking activity in both parent and child.

Or is there any mechanism where I can associate my object with the workflow to view the status of workflow instance against that object?

Thanks

**Comment by sfmskywalker** [comment]
Should be possible. When you run a child workflow while blocking, the parent workflow knows about the child workflow's instance ID. Additionally, or alternatively, you could run the child workflow with a correlation ID that you provide yourself from the parent workflow. Then you can use this correlation ID to lookup the child workflow instance.
